File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportPersistence/src/main/java/root/persistence/tickets/ITicketRepository.java
package root.persistence.tickets;

import root.model.Ticket;
import root.persistence.Repository;

import java.util.Collection;

public interface ITicketRepository extends Repository<Ticket, Integer> {
    Ticket findByTouristName(String touristName);
    Collection<Ticket> getByInvoiceId(int invoiceID);
    Collection<Ticket> getByFlightId(int flightID);
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/persistence/tickets/ITicketRepository.cs
using model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace persistence.tickets
{
    public interface ITicketRepository : Repository<Ticket, int>
    {
        Ticket? findByTouristName(string name);
        List<Ticket> getByInvoiceId(int invoiceId);
        List<Ticket> getByFlightId(int flightId);
    }
}
